tracing/src/macros.rs

```diff
@@ -983,12 +997,12 @@ macro_rules! enabled {
             let interest = CALLSITE.interest();
             if !interest.is_never() && CALLSITE.is_enabled(interest)  {
                 let meta = CALLSITE.metadata();
-                $crate::dispatch::get_default(|current| current.enabled(meta))
+                $crate::dispatch::get_default(|current| current.enabled(meta)) || $crate::log_enabled!(target: $target, $lvl)
```

Comment:
I was not sure how to simulate this situation in a test case. under what situations would `!interest.is_never() && CALLSITE.is_enabled(interest)` be true, but `$crate::dispatch::get_default(|current| current.enabled(meta))` be false?

Reply:
> under what situations would `!interest.is_never() && CALLSITE.is_enabled(interest)` be true, but `$crate::dispatch::get_default(|current| current.enabled(meta))` be false?

This would be the case any time a span or event is filtered dynamically (such as based on the current span).